src/bun.js/ipc.zig

```diff
@@ -156,17 +156,162 @@ pub const IPCData = struct {
     }
 };
 
-/// This type is shared between VirtualMachine and Subprocess for their respective IPC handlers
-///
-/// `Context` must be a struct that implements this interface:
-/// struct {
-///     globalThis: ?*JSGlobalObject,
-///     ipc: IPCData,
-///
-///     fn handleIPCMessage(*Context, DecodedIPCMessage) void
-///     fn handleIPCClose(*Context, Socket) void
-/// }
-pub fn NewIPCHandler(comptime Context: type) type {
+const NamedPipeIPCData = struct {
+    const uv = bun.windows.libuv;
+
+    pipe: uv.uv_pipe_t,
+    incoming: bun.ByteList = .{}, // Maybe we should use IPCBuffer here as well
+    outgoing: IPCBuffer = .{},
+    connected: bool = false,
+    has_written_version: if (Environment.allow_assert) u1 else u0 = 0,
+    connect_req: uv.uv_connect_t = std.mem.zeroes(uv.uv_connect_t),
+    server: uv.uv_pipe_t = std.mem.zeroes(uv.uv_pipe_t),
+    current_payload_len: usize = 0,
+
+    pub fn processSend(this: *NamedPipeIPCData) void {
+        const bytes = this.outgoing.list.slice()[this.outgoing.cursor..];
+        log("processSend {d}", .{bytes.len});
+        if (bytes.len == 0) return;
+
+        const req = bun.new(uv.uv_write_t, std.mem.zeroes(uv.uv_write_t));
+        req.data = @ptrCast(this);
+        req.write_buffer = uv.uv_buf_t.init(bytes);
+        log("processSend write_buffer {d}", .{req.write_buffer.len});
+        this.current_payload_len = bytes.len;
+        const write_err = uv.uv_write(req, @ptrCast(&this.pipe), @ptrCast(&req.write_buffer), 1, NamedPipeIPCData.uvWriteCallback).int();
```

Comment:
Why are we calling `uv.uv_write` directly? Can you add methods to `this.pipe` to make this code more maintainable?